### ISSUE TITLE
Invalid Data on Commit for Edited Match ("When only 1 robot plays, ...")

### ISSUE THREAD

**Issue by messenger-sean** [FIRST-Tech-Challenge/scorekeeper#458]
**Problem:** When trying to edit a match to add "Parked" for endgame for two robots, we receive an error "Invalid Data" when Committing.

**Details:** This is Qualification 13 for FIM Oxford 1 (https://theorangealliance.org/matches/1920-FIM-OWQ1-Q013-1). The match as posted at the time of writing includes only 1 Parked, as we were unable to Park both robots. However, both robots should be Parked.

When trying to add Parked on both robots, we receive an error that says "Invalid Data: When only 1 robot plays, robot 1 and robot 2 cannot both have robot-specific scoring achievements (navigation, parking, capstone). Check red." 

None of the robots were listed as No-Show. The scores were originally submitted with both robots showing Present and neither Parked.

**Images:** 

These are what we would _like_ to save and commit in match edit.

![IMG_20191123_120141](https://user-images.githubusercontent.com/4318057/69482973-fbbc7c00-0def-11ea-90e6-c3f8694eed44.jpg)

![IMG_20191123_120147](https://user-images.githubusercontent.com/4318057/69482976-01b25d00-0df0-11ea-8dde-ce27db320be6.jpg)

![IMG_20191123_120151](https://user-images.githubusercontent.com/4318057/69482979-13940000-0df0-11ea-859c-c8487c09f708.jpg)

**Comment by messenger-sean** [comment]
Notably, this is reproducible after several matches have been played after this match under question as well.

**Comment by adwiii** [comment]
Yes, thank you for bringing this up. This is a breaking bug in v1.2.0, and we have pulled down the release of v1.2.0. When using Live Scoring in v1.2.0, the "Commit" button on the Edit Scores tab will not function immediately after the match ends. 

Although it does not address matches that played but not committed, we recommend reverting to v1.1.x or v1.2.1 (coming soon). See below for instructions on how to get around this.

To avoid this happening, always use the "Commit and Post Last Match" button, even if the scores need to be edited. Once they have been committed this way, the scores can be edited and re-committed and re-posted.


If you have already hit this issue with a match:
## Viewing saved scores
The data is backed up. To see the un-committed results, navigate to the Match Results page, and select the "View Match History" link. 
![image](https://user-images.githubusercontent.com/9453262/69483154-e1839d80-0df1-11ea-804f-ebbd3d402651.png)

Then select the most recent edit and click "Scoresheet"
![image](https://user-images.githubusercontent.com/9453262/69483160-ef392300-0df1-11ea-89f2-2a22586f4078.png)

![image](https://user-images.githubusercontent.com/9453262/69483365-68d21080-0df4-11ea-9933-770fe7352c2b.png)

## Repairing matches
1. On the Match Control Page Schedule tab, press "Replay" for the match. This will clear the results from the first time the match was played. Then, press "Edit". You can then re-enter the scores from the scoresheets in the Match History.
2) Restart the scoring system. The Match Control Page will show the match as if it is unplayed, and you can enter the scores from the scoresheets in the Match History



We are working on posting a fix for this if swapping to v1.2.1 is feasible, However, the matches that have already been affected will still need to be manually repaired. Please feel free to post here with any additional concerns. We can assist to repair databases that are posted here. Please do not upload the db to _FIRST_ until this issue has been resolved. We are sorry for the inconvenience.

**Comment by tvbarnette999** [comment]
Additional note, if you are running a Qualifier and run into this, do NOT start Alliance Selection until all matches have been repaired AND COMMITTED.

**Comment by kenpugsley** [comment]
Seems as though the "Commit and Post Last Match" and the "Replay" match workaround may not always work.  Restarting the scoring software does seems to always allow editing of the match.

The following situation seems to will cause the issue  ...

1.  Live Scoring Using Tablets
2. Run a match, only 1 team on an alliance is scores with Navigated
3. After match, use "Commit and Post Last Match"
4. From scorekeeper computer edit the match and give both teams on the alliance credit for parking.  The error message (incorrectly) comes up and blocks entry.
4a. The workaround of Reply of the match in this case doesn't seem to resolve the issue either. 

As noted above, a restart of the scoring software does resolve the issue, though of course does require all screens/tablets to log back in.

Note:  In our tests, the issue seems to be limited to the fields in the error message ("Navigated", "Capstone", "Parked"), and only when editing to give both teams on an alliance those scores.

**Comment by adwiii** [comment]
In v1.3.0 more robust protections have been added to address the originally intended feature that caused this bug. In a 2 team match (qualifications or 2 team eliminations), if 1 robot is marked as a no show then there can only be 1 robot marked for robot-specific scoring achievements (navigating, parking, capping). If both robots are marked as no show, then no robot-specific scoring achievements can be marked. Similarly, in three team alliances for eliminations, if 2 robots are marked as no show then only 1 robot can have robot-specific scoring achievements and if all 3 robots are marked as no show then no robot-specific scoring achievements can be marked. 

Because the way the system calculates which robot is robot 1 is not always obvious, the system will accept either robot for the achievements. For example, even though the robot in the second spot below is marked as a no show, the system accepts inputting robot 2 as navigated, parked, and capstone scored at level 0.
![image](https://user-images.githubusercontent.com/9453262/72025908-ac1efc00-3247-11ea-8d9b-b872fb5ad183.png)

However, marking robot 1 as parked instead is not allowed because now both robots are marked as scoring.

![image](https://user-images.githubusercontent.com/9453262/72025970-e5f00280-3247-11ea-9ddf-f94c0816f4ce.png)
